src/layouts/Sidebar.jsx

```diff
@@ -39,6 +41,55 @@ const Sidebar = ({ menu, activeMenuItem }) => {
     setActiveState({ ...activeState, sub: subLinkId });
   };
 
+  useEffect(() => {
+    axios.get(`${API_END_POINT}/api/list/batch/`, { headers }).then((res) => {
+      setBatchList(res.data.data);
+    });
+  }, []);
+
+  const [showInputFields, setShowInputFields] = useState(false);
+
+  const [isAddingBatch, setIsAddingBatch] = useState(false); // New state for switching text
+
+  const toggleInputFields = () => {
+    setShowInputFields((prevState) => !prevState); // Toggle input fields visibility directly using prevState
+    setIsAddingBatch(!showInputFields);
+
+    // setBatchList(true)// Update text directly based on the current state of input fields
+  };
+
+  // const [loading, setLoading] = useState(false);
+  // const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const [batch_name, setBatchName] = useState("");
+  const [start_date, setStartDate] = useState("");
+  const [end_date, setEndDate] = useState("");
+  const [company, setCompany] = useState(1);
+
+  const [startError, setStartError] = useState("");
+  const [endError, setEndError] = useState("");
+  const [batchNameError, setBatchNameError] = useState(null);
+
+  const [isEditMode, setIsEditMode] = useState(false);
+  const [selectedBatch, setSelectedBatch] = useState(null);
+
+ 
+
+  const handleSwitch = (id, batchName) => {
+    Modal.confirm({
+      title: `Confirm Switch to ${batchName}`,
+      content: "Are you sure you want to Switch this Batch?",
+      onOk: () => {
+        navigate(`/batch/${id}/applications`);
+        window.location.reload();
```

Comment:
We find a way to load the value the current batch with without window.location.reload() we'll do it after demo.
Make note of it @Ganeshdckap